Result/Result.swift

```diff
@@ -179,6 +179,15 @@ public func ?? <T, Error> (left: Result<T, Error>, @autoclosure right: () -> Res
 	return left.recoverWith(right())
 }
 
+// MARK: - Derive result from failable closure
+
+public func materialize<T, E: ErrorType>(f: () throws -> T) -> Result<T, E> {
+    do {
+        return .success(try f())
+    } catch {
+        return .failure(error as! E)
```

Comment:
Is `as!` (forced) required? For some reason you can always do `errorType as NSError` and that always works :-/
If this is going to be a forced-casting, perhaps I would provide a separate method that just gives you an `ErrorType`?